src/pilot/assign_role.py

```diff
@@ -1161,6 +1166,23 @@ def select_os_volume(os_volume_size_gb, ironic_client, drac_client, node_uuid):
                     break
 
             if raid_size_gb == 0:
+                try:
+                    physical_disk_view_doc = drac_client.enumerate(
+                        DCIM_PhysicalDiskView)
+                    physical_disk_docs = utils.find_xml(
+                        physical_disk_view_doc,
+                        'DCIM_PhysicalDiskView',
+                        DCIM_PhysicalDiskView,
+                        True)
+                    disks = [get_size_in_bytes(physical_disk_doc,
+                                               DCIM_PhysicalDiskView)
+                             for physical_disk_doc in physical_disk_docs]
+                    os_volume_size_gb = int(disks[0]) / units.Gi
+                except:
+                    raise RuntimeError(
+                        "There must be either a virtual disk that "
+                        "is not a RAID 0 to install the OS on, or "
+                        "os-volume-size-gb must be specified")
                 raise RuntimeError("There must be either a virtual disk that "
```

Comment:
I don't see how this could possibly work because the above raise will be executed immediately following the try/except block, which means that an exception will always be thrown in the case where there is a jbod disk and no virtual disk.

Reply:
Yes i have removed already.